### DISCUSSION TITLE
winxp setup : sanbootconf.sys  not found

### DISCUSSION THREAD

**Discussion by 90999** [ipxe/ipxe#622] [General]
https://ipxe.org/appnote/xp_2003_direct_install

guest:  VMWARE E1000  without romfile ,  tiny pxe server  with ipex.iso 
ipex script:
#!ipxe
dhcp net0
sanboot iscsi:192.168.1.10::::iqn.1991-05.com.microsoft:target1

setp 10  not go on:
the file  sanbootconf.sys  could not be found   

anybody can help me ?

**Comment by NiKiZe** [comment]
Assuming typos and that ipex should be iPXE.

I have a hard time understanding which message you have where, but maybe these sites are helpful:
* https://ipxe.org/sanbootconf
* https://github.com/ipxe/sanbootconf

**Reply by 90999** [reply]
> Assuming typos and that ipex should be iPXE.
> 
> I have a hard time understanding which message you have where, but maybe these sites are helpful:
> 
>     * https://ipxe.org/sanbootconf
> 
>     * https://github.com/ipxe/sanbootconf

the same error

**Reply by stappersg** [reply]
Then describe your error better. Do something with the _I have a hard time understanding which message you have where_.

**Reply by 90999** [reply]
new reply at https://github.com/ipxe/ipxe/discussions/622  ，thanks for your reply

**Comment by 90999** [comment]
that's what i do :

1. build a vhd file , mount to HD(IDE) in vmware , run the winpe ,  active partition  and format to fat32 
2. change cdrom to winxp.iso , run D:\I386\winnt32.exe /syspart:c /tempdrive:c /makelocalsource /noreboot 
3. when step.2 done,  run IntegrateDrv.exe /driver=..\Drivers\E1000\x86 /driver=..\Drivers\iScsiPrt\x86 /driver=..\Drivers\sanbootconf /driver=..\Drivers\nicbootconf /target=C:\ , 
4. select "25" for network adapter driver,  IP address is default (enter buttom)
5. unmount vhd in vmware , mount to iscsi server , setting  tiny ipex server , select httpd and proxy dhcp , boot file is ipex.iso
6. select “if user-class =ipex“ ，scripts file is  “dhcp net0” and “sanboot iscsi:192.168.1.10::::iqn.1991-05.com.microsoft:target1”
7. vmware running without cdrom and hddriver ,select E1000 boot,  after i can see the winxp setup is loading some drivers
8. wait a sec，it told me “ the file sanbootconf.sys could not be found ” and setup cannot go on .


Normally ， setp.8 is  “The installation should detect your iSCSI disk” ，the same as step10 at https://ipxe.org/appnote/xp_2003_direct_install  ，but it doesnot . when the ipex.iso booting ， i can see  a client connect the iscsi server  until winxp setup.
i try two winxp version sp2 and sp3 , and i copy sanbootconf.sys to every folds， the result is still the same 。

PS：
i not Configure the vm ， like https://ipxe.org/howto/vmware ， WIN7  setup is working good 。
![1](https://user-images.githubusercontent.com/31765469/160803815-2f196c6d-660c-4887-991b-f97b02c103cb.jpg)

**Reply by stappersg** [reply]
I could not resist to express  _it's not worth my time _

![image](https://user-images.githubusercontent.com/35434094/160811242-343c64ba-d469-4438-b992-f3dfc1b0060f.png)

**Reply by NiKiZe** [reply]
Could you take some time and make sure that what you are posting is correct? Go over and make sure that you have ipxe in the correct places in your files and not something else.

Check that the file exists and is in the correct place relative to the inf file.

**Reply by 90999** [reply]
i am sure that I followed the link step by step, the script is correct, it loads the ISCSI disk and boots as I wrote it, 
but execution stops at sanbootconf loading. 

I used the a script to load WIN10.iso, and found that the ISCSI can be recognized as the E drive, and then I can install it on it.

script :

#!ipxe
dhcp net0
set net0/gateway 0.0.0.0
set keep-san 1
sanhook --drive 0x81 iscsi:192.168.1.11::::iqn.1991-05.com.microsoft:target1
echo 4   iqn.1991-05.com.microsoft:target1
sanboot --drive 0xff http://192.168.1.11/x.iso


I think because the sanbootconf driver was not properly installed on the winxp setup, either my steps were wrong, 
or sanbootconf is too new to unsupport winxp. 


![2](https://user-images.githubusercontent.com/31765469/161116953-344c4fac-0902-4f76-bee4-5b17ec1aecb5.jpg)

**Reply by 90999** [reply]
weekend,  i will try the "Method #2: Creating an iSCSI capable installation CD " to install winxp .

**Reply by 90999** [reply]
now  winxp step is working use   " iSCSI capable installation CD "

**Reply by NiKiZe** [reply]
Following step by step might not always be enough, you might need to do corrections. The best is if you understand what the directions do and why.

**Reply by 90999** [reply]
nooooooooooooooooooooooooo!!!!!!!!!!!!!
iSCSI capable installation CD   pre install  is working , 

but 

sanboot iscsi XXXXXXXX  

is not working too.
it's no " iscsi boot via ipex"
i think  the nic , iscsi , sanbootconf driver are not loading  before  GUI-mode installation , blue screen  0x0000007F  

winxp diskledd is so hard ? 


step10.
![1](https://user-images.githubusercontent.com/31765469/161256936-6db75c37-9db8-4463-817d-ca8e217414c5.png)


![1 1](https://user-images.githubusercontent.com/31765469/161256954-2800684a-0215-49ad-8200-bb7f78e5c186.jpg)

reboot,  blue screen  0x0000007F

**Reply by 90999** [reply]
ipex script:
dhcp net0

#set net0/gateway 0.0.0.0
#set keep-san 1
#it's not need two cmd 

#sanhook iscsi:192.168.1.11::::iqn.1991-05.com.microsoft:target1 
#first time use sanhook iscsi with winxp iscsi cd ,after use sanbook iscsi 

sanboot iscsi:192.168.1.11::::iqn.1991-05.com.microsoft:target1

now the gui-mode install is working ............

**Reply by 90999** [reply]
i don't know what happen when i use Method 1 ， but Method 2 is wokring good。

now i try to add some  driver  to no iscsi support xp system ,copy  partition to vhd ,and boot with ipxe
